src/NServiceBus.Core/Pipeline/ConnectorContextExtensions.cs

```diff
@@ -218,19 +218,19 @@ public static IOutgoingPhysicalMessageContext CreateOutgoingPhysicalMessageConte
         /// <summary>
         /// Creates a <see cref="IAuditContext" /> based on the current context.
         /// </summary>
-        public static IAuditContext CreateAuditContext(this ForkConnector<IIncomingPhysicalMessageContext, IAuditContext> forkConnector, OutgoingMessage message, string auditAddress, IIncomingPhysicalMessageContext sourceContext)
+        public static IAuditContext CreateAuditContext(this ForkConnector<IIncomingPhysicalMessageContext, IAuditContext> forkConnector, OutgoingMessage message, string auditAddress, TimeSpan? timeToBeReceived, IIncomingPhysicalMessageContext sourceContext)
```

Comment:
This method is public. Would we need to evolve it by adding an overload and obsolete the old?

By the way the Azure SDK usually then also adds EditorBrowsable State never to the old one which hides the method from intellisense

Reply:
Good point, I've added both the obsolete and the EditorBrowsable attribute

Reply:
If you like this trick we can add it to the breaking changes guidelines. What do you think?

Reply:
Yea, I'll do a PR